app/services/support_interface/application_choices_export.rb

```diff
@@ -68,5 +69,24 @@ def relevant_applications
         .where.not(submitted_at: nil)
         .order('submitted_at asc')
     end
+
+    def format_structured_rejection_reasons(structured_rejection_reasons)
+      return nil if structured_rejection_reasons.blank?
+
+      select_high_level_rejection_reasons(structured_rejection_reasons)
+      .keys
+      .map { |reason| format_reason(reason) }
+      .join("\n")
+    end
+
+    def select_high_level_rejection_reasons(structured_rejection_reasons)
+      structured_rejection_reasons.select { |reason, value| value == 'Yes' && reason.include?('_y_n') }
```

Comment:
All the high-level rejection reasons have the `reason_y_n` format, so this allows us to filter based on that pattern

Reply:
I think it would be nice to have a `top_level_reason?` helper method somewhere. Then we could start to tidy up some of these 'metadata lookups'. You use `SupportInterface::SubReasonsForRejectionTableComponent::TOP_LEVEL_REASONS_TO_I18N_KEYS.keys` but that's not really the right places for the R4R model either.